src/jsrs_parser.cc

```diff
@@ -74,28 +76,30 @@ static constexpr Local<Value> (*kParseFunctions[])(Isolate*,
 };
 
 Local<Value> Parse(Isolate* isolate, const String::Utf8Value& in) {
-  size_t size;
-  const char* to_parse =
-      internal::PrepareString(isolate, *in, in.length(), &size);
-  if (!to_parse) {
-    return Undefined(isolate);
-  }
+  const char* str = *in;
+  const size_t length = in.length();
+  const char* end = str + length;
 
   Type type;
-  if (!GetType(to_parse, to_parse + size, &type)) {
+
+  size_t start_pos = internal::SkipToNextToken(str, end);
+  if (!GetType(str + start_pos, end, &type)) {
     THROW_EXCEPTION(TypeError, "Invalid type");
     return Undefined(isolate);
   }
 
   size_t parsed_size = 0;
   Local<Value> result =
-      (kParseFunctions[type])(isolate, to_parse, to_parse + size, &parsed_size);
-  if (size != parsed_size) {
+      (kParseFunctions[type])(isolate, str + start_pos, end, &parsed_size);
+
+  parsed_size += internal::SkipToNextToken(str + start_pos + parsed_size, end);
```

Comment:
Why not increment `str` pointer after each `SkipToNextToken` call? We don't need the initial address anyway.